wimm.Secundatives/NaturalRange.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+
+namespace wimm.Secundatives
+{
+    /// <summary>
+    /// A <see cref="Range{T}"/> of the natural numbers.
+    /// </summary>
+    /// <inheritDoc />
+    public class NaturalRange : IRange<int>
+    {
+        private const string _outOfRangeMessage = "Must be greater than or equal to zero.";
+        private readonly Range<int> _range;
+
+        public int Max => _range.Max;
+
+        public int Min => _range.Min;
+
+        /// <summary>
+        /// Initializes a new <see cref="NaturalRange"/> of 0 to <paramref name="max"/>.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="max"/> is less than zero.
+        /// </exception>
+        public NaturalRange(int max) : this(0, max) { }
+
+        /// <summary>
+        /// Initializes a new <see cref="NaturalRange"/>.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="min"/> or <paramref name="max"/> is less than zero.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="max"/> is less than <paramref name="min"/>.
+        /// </exception>
+        public NaturalRange(int min, int max)
```

Comment:
Is there a reason that you're not just making NaturalRange : Range<int> ?

Reply:
I was going to but then constructor chaining order would put the `min <= max` before the `x >= 0` validation. Of the things I considered; base class vs interface only, exception precedence, and extra code/behavior duplication, the only one that had public facing implications I cared about was exception precedence so that's what I favored. That said, they're all minor trade offs so ICBC to change it.